easybuild/easyblocks/x/xmipp.py

```diff
@@ -25,174 +25,203 @@
-            'profile=no fast=yes warn=no release=yes gtest=yes static=no cuda=no debug=no matlab=no',
-            'LINKERFORPROGRAMS=%s' % os.getenv('CXX'),
-            'MPI_BINDIR=%s' % mpi_bindir,
-            'MPI_LIB=mpi',
-            'JAVA_HOME=%s' % os.getenv('JAVA_HOME'),
-            'JAVAC=javac',
-            'CC=%s' % os.getenv('CC'),
-            # pass $CXXFLAGS in Python list syntax and avoid spaces, e.g.: ['-O2','-march=native']
-            'CXXFLAGS=%s' % str(os.getenv('CXXFLAGS').split(' ')).replace(' ', ''),
-            'CXX=%s' % os.getenv('CXX'),
-            'MPI_CC=%s' % os.getenv('MPICC'),
-            # pass $CFLAGS in Python list syntax and avoid spaces, e.g.: ['-O2','-march=native']
-            'CCFLAGS=%s' % str(os.getenv('CFLAGS').split(' ')).replace(' ', ''),
-            'MPI_CXX=%s' % os.getenv('MPICXX'),
-            'MPI_INCLUDE=%s' % os.getenv('MPI_INC_DIR'),
-            'MPI_LIBDIR=%s' % os.getenv('MPI_LIB_DIR'),
-            'MPI_LINKERFORPROGRAMS=%s' % os.getenv('MPICXX'),
-            'LIBPATH=%s' % os.getenv('LD_LIBRARY_PATH'),
-        ])
-
-        # define list of configure options, which will be passed to Xmipp's install.sh script via --configure-args
-        self.cfg['configopts'] = configure_args
-        self.log.info("Configure arguments for Xmipp install.sh script: %s", self.cfg['configopts'])
+        """Custom configuration procedure for Xmipp."""
+
+        # Initialize the config file and then patch it with the correct values
+        self.cfgfile = os.path.join(self.builddir, 'xmipp.conf')
+        cmd = '%s src/xmipp/xmipp config' % self.cfg['preconfigopts']
+        run_cmd(cmd, log_all=True, simple=True)
+
+        params = {
+            'CC': os.environ['CC'],
+            'CXX': os.environ['CXX'],
+            'LINKERFORPROGRAMS': os.environ['CXX'],
+            'LIBDIRFLAGS': '',
+            # I don't think Xmipp can actually build without MPI.
+            'MPI_CC': os.environ.get('MPICC', 'UNKNOWN'),
+            'MPI_CXX': os.environ.get('MPICXX', 'UNKNOWN'),
+            'MPI_LINKERFORPROGRAMS': os.environ.get('MPICXX', 'UNKNOWN'),
+        }
+
+        deps = [
+            # dep name, is required dep?
+            ('HDF5', None, True),
+            ('SQLite', None, True),
+            ('LibTIFF', None, True),
+            ('libjpeg-turbo', None, True),
+            ('Java', 'JAVA_HOME', True),
+            ('MATLAB', 'MATLAB_DIR', False),
+        ]
+
+        regex_subs = []
+
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            params.update({'CUDA_BIN': os.path.join(cuda_root, 'bin')})
+            params.update({'CUDA_LIB': os.path.join(cuda_root, 'lib64')})
+            params.update({'NVCC': os.environ['CUDA_CXX']})
+            # Their default for NVCC is to use g++-5, fix that
+            nvcc_flags = '--x cu -D_FORCE_INLINES -Xcompiler -fPIC '
+            nvcc_flags = nvcc_flags + '-Wno-deprecated-gpu-targets -std=c++11'
```

Comment:
Aren't we going to regret hardcoding the `std=c++11` later?

Reply:
Maybe, I just didn't feel like changing that from what they use at the moment.

Reply:
OK, fine, we'll see